src/System.Threading.Tasks/tests/System.Runtime.CompilerServices/AsyncTaskMethodBuilderTests.netcoreapp.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Xunit;
+
+namespace System.Threading.Tasks.Tests
+{
+    public partial class AsyncTaskMethodBuilderTests : RemoteExecutorTestBase
+    {
+        [OuterLoop]
+        [Fact]
+        public static void DroppedIncompleteStateMachine_RaisesIncompleteAsyncMethodEvent()
+        {
+            RemoteInvoke(() =>
+            {
+                using (var listener = new TestEventListener("System.Threading.Tasks.TplEventSource", EventLevel.Verbose))
+                {
+                    var events = new ConcurrentQueue<EventWrittenEventArgs>();
+                    listener.RunWithCallback(events.Enqueue, () =>
+                    {
+                        NeverCompletes();
+                        GC.Collect();
+                        GC.WaitForPendingFinalizers();
```

Comment:
I believe there is currently some sort of race with WaitForPendingFinalizers() where if a GC had occurred prior to Collect() above and the finalizer thread is almost done, the first call to Wait may correspond to the first GC and not the Collect() above. To work around this I had previously added a second call to Wait, that seemed to resolve the issue. IIRC some of the GC tests also use a similar workaround.

Reply:
Added.